src/sqlfluff/core/rules/std.py

```diff
@@ -3244,3 +3244,60 @@ def _eval(self, segment, **kwargs):
                 self.violation_buff[-1].fixes = fixes
 
         return self.violation_buff or None
+
+
+@std_rule_set.document_fix_compatible
+@std_rule_set.register
+class Rule_L035(BaseCrawler):
+    """Do not specify "else null" in a case when statement (redundant).
+
+    | **Anti-pattern**
+
+    .. code-block:: sql
+
+        select
+            case
+                when name like '%cat%' then 'meow'
+                when name like '%dog%' then 'woof'
+                else null
+            end
+        from x
+
+
+    | **Best practice**
+    |  Omit "else null"
+
+    .. code-block:: sql
+
+        select
+            case
+                when name like '%cat%' then 'meow'
+                when name like '%dog%' then 'woof'
+            end
+        from x
+
+    """
+
+    def _eval(self, segment, **kwargs):
+        if segment.is_type("case_expression"):
+            fixes = []
+            for idx, seg in enumerate(segment.segments):
+                # When we find ELSE we delete
+                # everything up to NULL
+                if fixes:
+                    fixes.append(LintFix("delete", seg))
+                    if seg.raw_upper == "NULL":
+                        # When we find NULL we're done
+                        return LintResult(anchor=segment, fixes=fixes)
```

Comment:
It's not immediately obvious, but I think worth a comment here, that it *should* be relatively safe to look for a `NULL` child segment here without accidentally picking up an `ExpressionSegment`, because if there was anything more complicated than just a bare `NULL` it would be contained within a child segment, rather than on it's own (and therefore while the `raw_upper` would _contain_ `NULL`, it wouldn't _equal_ `NULL`).

Reply:
Ah good point! I added a test for that "expression in ELSE" case just to be sure